common/lib/xmodule/xmodule/modulestore/xml_importer.py

```diff
@@ -739,11 +740,27 @@ def _convert_ref_fields_to_new_namespace(reference):  # pylint: disable=invalid-
     fields = _update_module_references(module, source_course_id, dest_course_id)
     asides = module.get_asides() if isinstance(module, XModuleMixin) else None
 
-    return store.import_xblock(
+    block = store.import_xblock(
         user_id, dest_course_id, module.location.category,
         module.location.block_id, fields, runtime, asides=asides
     )
 
+    # TODO: Move this code once the following condition is met.
+    # Get to the point where XML import is happening inside the
+    # modulestore that is eventually going to store the data.
+    # Ticket: https://openedx.atlassian.net/browse/PLAT-1046
+    if block.location.category == 'library_content':
+        # if library exists, update source_library_version and children
+        # according to this existing library and library content block.
+        if store.get_library(block.source_library_key):
+            LibraryToolsService(store).update_children(
```

Comment:
Now this change makes sense, previously if publish branch, library children were updated for both branches and a new structure was made, causing the the issue when we export a published version of the content against the same draft version of that content.